connectors/src/connectors/github/lib/github_api.ts

```diff
@@ -867,10 +867,7 @@ export async function processRepository({
         `${path.join("/")}/${fileName}`
       );
 
-      const parentInternalId =
-        parents.length === 0
-          ? null
-          : (parents[parents.length - 1]?.internalId as string);
+      const parentInternalId = parents[0]?.internalId ?? null;
```

Comment:
nit: here we can explode if parents[0] is set but parents[0].internalId is not set no?

Reply:
indeed, that would be a sneaky error too but I think we're good since `getCodeDirInternalId` never returns null

Reply:
yes I'm convinced it won't happen too but the way we usually deal with that to please typescript is to throw unreachable